cassandra-diagnostics-core/src/main/java/io/smartcat/cassandra/diagnostics/module/heartbeat/HeartbeatModule.java

```diff
@@ -21,38 +24,29 @@
 
     private static final Logger logger = LoggerFactory.getLogger(HeartbeatModule.class);
 
-    private static final String PERIOD_PROP = "period";
-
-    private static final String DEFAULT_PERIOD = "15";
-
-    private static final String TIMEUNIT_PROP = "timeunit";
-
-    private static final String DEFAULT_TIMEUNIT = "MINUTES";
+    private final HeartbeatConfiguration options;
 
     private final String service;
 
-    private final int period;
-
-    private final TimeUnit timeunit;
-
     private final Timer timer;
 
     /**
      * Constructor.
      *
      * @param configuration Module configuration
      * @param reporters Reporter list
+     * @throws ConfigurationException in case the provided module configuration is not valid
      */
-    public HeartbeatModule(ModuleConfiguration configuration, List<Reporter> reporters) {
+    public HeartbeatModule(ModuleConfiguration configuration, List<Reporter> reporters) throws ConfigurationException {
         super(configuration, reporters);
 
-        period = Integer.parseInt(configuration.getDefaultOption(PERIOD_PROP, DEFAULT_PERIOD));
-        timeunit = TimeUnit.valueOf(configuration.getDefaultOption(TIMEUNIT_PROP, DEFAULT_TIMEUNIT));
+        options = HeartbeatConfiguration.create(configuration.options);
         service = configuration.measurement;
 
-        logger.debug("Heartbeat module initialized with {} period and {} timeunit.", period, timeunit.name());
+        logger.debug("Heartbeat module initialized with {} period and {} timeunit.",
+                options.period(), options.timeunit().name());
         timer = new Timer();
-        timer.schedule(new HeartbeatTask(), timeunit.toMillis(period));
+        timer.schedule(new HeartbeatTask(), options.timeunit().toMillis(options.period()));
```

Comment:
Can this be internal method of options, something like periodInMillis or something? It looks to me it should be internal method, not done here